Processor/ServicesResetter/ServicesResetterProcessor.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Swarrot\SwarrotBundle\Processor\ServicesResetter;
+
+use Swarrot\Broker\Message;
+use Swarrot\Processor\ProcessorInterface;
+use Swarrot\Processor\SleepyInterface;
+use Symfony\Contracts\Service\ResetInterface;
+
+class ServicesResetterProcessor implements SleepyInterface
+{
+    /**
+     * @var ProcessorInterface
+     */
+    private $processor;
+
+    /**
+     * @var ResetInterface
+     */
+    private $servicesResetter;
+
+    public function __construct(ProcessorInterface $processor, ResetInterface $servicesResetter)
+    {
+        $this->processor = $processor;
+        $this->servicesResetter = $servicesResetter;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(Message $message, array $options)
+    {
+        return $this->processor->process($message, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sleep(array $options)
+    {
+        $this->servicesResetter->reset();
```

Comment:
This is convenient to reset services here to ensure that all processors have done their job. But should be better to do it in process() and write a documentation about the processor to explain that it should be the last processor to be executed in the stack. What do you think ?

Reply:
The `sleep` method is called only when no message is found in the broker.
If you have a bunch of messages to process, the `servicesResetter` won't be called between all messages.

Reply:
See this Doctrine processor to see how it's done: https://github.com/swarrot/swarrot/blob/master/src/Swarrot/Processor/Doctrine/ObjectManagerProcessor.php

Reply:
OK, I will do that